sql-plugin/src/main/spark332db/scala/org/apache/spark/sql/execution/datasources/GpuWriteFiles.scala

```diff
@@ -196,4 +199,24 @@ object GpuWriteFiles {
       case w: GpuWriteFilesExec => w
     }
   }
+
+  def getHadoopTaskAttemptContext(hadoopConf: Configuration): TaskAttemptContext = {
```

Comment:
nit: the method name and signature make it seem that `HadoopTaskAttemptContext` is extracted from hadoopConf. But it is almost the opposite: hadoopConf is updated with ids created from a Spark task context